packages/e2e-test-app-fabric/windows/RNTesterApp-Fabric/RNTesterApp-Fabric.cpp

```diff
@@ -0,0 +1,283 @@
+      m_host.InstanceSettings(InstanceSettings());
+    }
+
+    return m_host;
+  }
+  winrt::Microsoft::ReactNative::ReactInstanceSettings InstanceSettings() noexcept {
+    if (!m_instanceSettings) {
+      m_instanceSettings = winrt::Microsoft::ReactNative::ReactInstanceSettings();
+    }
+
+    return m_instanceSettings;
+  }
+
+  LRESULT RenderApp(HWND hwnd) {
+    PCWSTR appName = L"RNTesterApp";
+
+    WCHAR workingDir[MAX_PATH];
+    GetCurrentDirectory(MAX_PATH, workingDir);
+
+    auto host = Host();
+    // Disable until we have a 3rd party story for custom components
+    // RegisterAutolinkedNativeModulePackages(host.PackageProviders()); // Includes any
+    // autolinked modules
+
+    host.InstanceSettings().JavaScriptBundleFile(m_bundleFile);
+
+    host.InstanceSettings().UseWebDebugger(m_useWebDebugger);
+    host.InstanceSettings().UseDirectDebugger(m_useDirectDebugger);
+    host.InstanceSettings().BundleRootPath(std::wstring(L"file:").append(workingDir).append(L"\\Bundle\\").c_str());
+    host.InstanceSettings().DebuggerBreakOnNextLine(m_breakOnNextLine);
+    host.InstanceSettings().UseFastRefresh(m_fastRefreshEnabled);
+    host.InstanceSettings().DebuggerPort(m_debuggerPort);
+    host.InstanceSettings().UseDeveloperSupport(true);
+
+    host.PackageProviders().Append(winrt::make<CompReactPackageProvider>());
+    winrt::Microsoft::ReactNative::ReactCoreInjection::SetTopLevelWindowId(
+        host.InstanceSettings().Properties(), reinterpret_cast<uint64_t>(hwnd));
+
+    // Nudge the ReactNativeHost to create the instance and wrapping context
+    host.ReloadInstance();
+
+    winrt::Microsoft::ReactNative::ReactViewOptions viewOptions;
+    viewOptions.ComponentName(appName);
+    m_CompositionHwndHost.ReactViewHost(
+        winrt::Microsoft::ReactNative::ReactCoreInjection::MakeViewHost(host, viewOptions));
+
+    auto windowData = WindowData::GetFromWindow(hwnd);
+    if (!windowData->m_windowInited) {
+      m_CompositionHwndHost.Initialize(static_cast<uint64_t>(hwnd));
+      windowData->m_windowInited = true;
+    }
+    return 0;
+  }
+
+  LRESULT TranslateMessage(UINT message, WPARAM wparam, LPARAM lparam) noexcept {
+    if (m_CompositionHwndHost) {
+      return static_cast<LRESULT>(m_CompositionHwndHost.TranslateMessage(message, wparam, lparam));
+    }
+    return 0;
+  }
+};
+
+extern "C" IMAGE_DOS_HEADER __ImageBase;
+HINSTANCE WindowData::s_instance = reinterpret_cast<HINSTANCE>(&__ImageBase);
+
+//
+//  FUNCTION: WndProc(HWND, UINT, WPARAM, LPARAM)
+//
+//  PURPOSE: Processes messages for the main window.
+//
+LRESULT CALLBACK WndProc(HWND hWnd, UINT message, WPARAM wParam, LPARAM lParam) {
+  auto windowData = WindowData::GetFromWindow(hWnd);
+  if (windowData) {
+    auto result = WindowData::GetFromWindow(hWnd)->TranslateMessage(message, wParam, lParam);
+    if (result)
+      return result;
+  }
+
+  switch (message) {
+    case WM_DESTROY: {
+      delete WindowData::GetFromWindow(hWnd);
```

Comment:
Kind of bypasses the agency of the unique_ptr. Whole process is about to terminate, so not sure how much I care. But feels rude 😅.

Is this here to ensure some destructors run first?

Reply:
This code is from playground-composition. @acoates-ms should have more context.